cmd/containerboot/kube_test.go

```diff
@@ -0,0 +1,206 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+//go:build linux
+
+package main
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"tailscale.com/kube"
+)
+
+func TestSetupKube(t *testing.T) {
```

Comment:
Some of this functionality is already tested in ./cmd/containerboot/main_test.go. This additonally tests error cases as well as serves as a documentation for which flows are supposed to succeed/fail on kube